MGSwipeTableCell/MGSwipeTableCell.m

```diff
@@ -831,9 +831,12 @@ -(void) refreshContentView
     CGFloat currentOffset = _swipeOffset;
     BOOL prevValue = _triggerStateChanges;
     _triggerStateChanges = NO;
-    self.swipeOffset = 0;
-    self.swipeOffset = currentOffset;
-    _triggerStateChanges = prevValue;
+    __block MGSwipeTableCell *blockSelf = self;
+    [self setSwipeOffset:0 animated:NO completion:^{
+        [blockSelf setSwipeOffset:currentOffset animated:NO completion:^{
+            _triggerStateChanges = prevValue;
```

Comment:
accessing this instance variable inside of the block also implicitly captures it as a strong reference. The recommended technique of accessing an instance variable like this inside a block is called "weak-strong dance". Where you use a weak reference as mentioned above and then convert it to a strong reference within the block and verify it is not nil before accessing the instance variable.